docs/adrs/0079-api-versioning.md

```diff
@@ -0,0 +1,124 @@
+---
+title: 0079 API Versioning
+description: How to implement API Versioning
+---
+
+# How to implement API Versioning for Prime API
+
+**User Story:** *[MB-15994](https://dp3.atlassian.net/browse/MB-15994)*
+
+## Problem Statement
+In order to support the Prime API Integration and prevent introducing potential breaking changes to the API, MilMove needs to version future updates to the Prime API. This ADR will attempt to address the "How?" questions associated with this endeavor. While some of the challenges involved are technical in nature, much of the challenge is coming up with patterns that are clear, understandable, and acceptable to the members of the engineering team.
+
+There are a few areas to consider when deciding how to implement API versioning:
+* [The overall versioning strategy (URL patterns, query params, headers, etc.)](#options-for-versioning-strategy)
+* [What endpoints will be incorporated into a new API version?](#codebase-management)
+* An overall file organization strategy - This could even be broken up into categories to consider on their own, because we don't necessarily want to take the same organization approach for all areas of the application
+  * [Swagger files](#swagger-organization)
+  * [Handlers](#handler-organization)
+  * [Services and fetchers](#servicesfetcher-organization)
+* [Documentation considerations](#documentation) - This is a necessity since it will be key ensuring the consumers of our API are informed on what is changed between versions
+
+## API Versioning Strategy
+
+### SemVer
+The Prime API follows the SemVer (Semantic Versioning) convention, using a three-part version number: Major.Minor.Patch. Here's what each part signifies:
+
+Major versions incorporate significant changes, including backward-incompatible API changes.
+
+Minor versions add new features in a backwards-compatible manner.
+
+Patch versions include backwards-compatible bug fixes.
+
+For example, a version could be noted as "2.1.3," where '2' is the major version, '1' is the minor version, and '3' is the patch version.
+
+### Options for Versioning Strategy
+There are a few options for implementing api versioning in our prime API.
+
+* Option 1: Include version in the URL patterns.
+  * Pros: This is the simplest option. We currently have `/v1/` in the paths of all of our APIs, and the implementation of our routing initialization can be easily modified to include multiple paths. If we went with some other option we would want to remove the `v1` from our outward facing APIs.
```

Comment:
I want to say this is easier to cache in some ways too, maybe more from a CDN/proxy perspective, than some of the other methods. There is a note here that gets into some of that https://learn.microsoft.com/en-us/azure/architecture/best-practices/api-design#media-type-versioning